components/scream/src/share/property_checks/mass_and_energy_conservation_check.cpp

```diff
@@ -0,0 +1,336 @@
+    const auto qv_i             = ekat::subview(qv, i);
+    const auto qc_i             = ekat::subview(qc, i);
+    const auto qr_i             = ekat::subview(qr, i);
+
+    energy(i) = compute_total_energy_on_column(team, pseudo_density_i, T_mid_i, horiz_winds_i,
+                                               qv_i, qc_i, qr_i, ps(i), phis(i));
+  });
+}
+
+PropertyCheck::ResultAndMsg MassAndEnergyConservationCheck::check() const
+{
+  auto mass   = m_current_mass;
+  auto energy = m_current_energy;
+
+  view_1d<Real> rel_err_mass  ("rel_err_mass",   m_num_cols);
+  view_1d<Real> rel_err_energy("rel_err_energy", m_num_cols);
+
+  EKAT_ASSERT_MSG(!std::isnan(m_dt), "Error! Timestep dt must be set in MassAndEnergyConservationCheck "
+                                     "before running check().");
+  auto dt = m_dt;
+
+  const auto pseudo_density = m_fields.at("pseudo_density")->get_view<const Real**> ();
+  const auto T_mid          = m_fields.at("T_mid"         )->get_view<const Real**> ();
+  const auto horiz_winds    = m_fields.at("horiz_winds"   )->get_view<const Real***>();
+  const auto qv             = m_fields.at("qv"            )->get_view<const Real**> ();
+  const auto qc             = m_fields.at("qc"            )->get_view<const Real**> ();
+  const auto qi             = m_fields.at("qi"            )->get_view<const Real**> ();
+  const auto qr             = m_fields.at("qr"            )->get_view<const Real**> ();
+  const auto ps             = m_fields.at("ps"            )->get_view<const Real*>  ();
+  const auto phis           = m_fields.at("phis"          )->get_view<const Real*>  ();
+
+  const auto vapor_flux = m_fields.at("vapor_flux")->get_view<const Real*>();
+  const auto water_flux = m_fields.at("water_flux")->get_view<const Real*>();
+  const auto ice_flux   = m_fields.at("ice_flux"  )->get_view<const Real*>();
+  const auto heat_flux  = m_fields.at("heat_flux" )->get_view<const Real*>();
+
+  const auto policy = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(m_num_cols, m_num_levs);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA (const KT::MemberType& team) {
+
+    const int i = team.league_rank();
+
+    const auto pseudo_density_i = ekat::subview(pseudo_density, i);
+    const auto T_mid_i          = ekat::subview(T_mid, i);
+    const auto horiz_winds_i    = ekat::subview(horiz_winds, i);
+    const auto qv_i             = ekat::subview(qv, i);
+    const auto qc_i             = ekat::subview(qc, i);
+    const auto qi_i             = ekat::subview(qi, i);
+    const auto qr_i             = ekat::subview(qr, i);
+
+    // Calculate total mass
+    const Real tm = compute_total_mass_on_column(team, pseudo_density_i, qv_i, qc_i, qi_i, qr_i);
+    const Real previous_tm = mass(i);
+
+    // Calculate expected total mass
+    const Real tm_exp = previous_tm +
+                        compute_mass_boundary_flux_on_column(vapor_flux(i), water_flux(i))*dt;
```

Comment:
Would you explain how you resolved the issue with dtime vs dtime/numsubcycles being used here?